src/native/managed/cdac/Microsoft.Diagnostics.DataContractReader.Legacy/TypeNameBuilder.cs

```diff
@@ -112,7 +112,7 @@ public static void AppendMethodImpl(Target target, StringBuilder stringBuilder,
                     arrayName = ".ctor";
                     break;
                 default:
-                    throw new ArgumentException(nameof(method));
+                    throw new ArgumentException(null, nameof(method));
```

Comment:
You should add some message here instead of passing null. Can be as simple as the following:
```suggestion
                    throw new ArgumentException($"Unkown ArrayFunctionType: {functionType}.", nameof(method));
```